dashboard/app/dsl/script_dsl.rb

```diff
@@ -82,19 +83,30 @@ def stage(name, properties = {})
     if @stage
       @stages << {
         stage: @stage,
+        visible_after: @stage_visible_after,
         scriptlevels: @scriptlevels,
         stage_extras_disabled: @stage_extras_disabled,
       }.compact
     end
     @stage = name
     @stage_flex_category = properties[:flex_category]
     @stage_lockable = properties[:lockable]
+    @stage_visible_after = determine_visible_after_time(properties[:visible_after])
     @scriptlevels = []
     @concepts = []
     @skin = nil
     @stage_extras_disabled = nil
   end
 
+  def determine_visible_after_time(visible_after_value)
+    current_time = Time.now
+    raw_diff_to_wed = 3 - current_time.wday
+    diff_to_wed = raw_diff_to_wed > 0 ? raw_diff_to_wed : 7 + raw_diff_to_wed
+    next_wednesday = current_time + diff_to_wed.day
+    next_wednesday_8_am_pst = Time.new(next_wednesday.year, next_wednesday.month, next_wednesday.day, 8, 0, 0, '-07:00').to_s
+    visible_after_value == '' ? next_wednesday_8_am_pst : visible_after_value
```

Comment:
the variable name says PST, but the offset is `-07:00` which is PDT. These should definitely be made consistent. I'd vote for sticking with PDT and just updating your variable name, so you don't have to change it in too many places, and also because that will be less unpleasant surprise in the unfortunate situation that we're still making new lessons like this in November.